File: router/example/task_config.go
package example

import (
	"github.com/gin-gonic/gin"
	v1 "github.com/xiaohongshu/PnSql/server/api/v1"
)

type TaskConfigRouter struct{}

func (s *SysUserRouter) InitTaskConfigRouter(Router *gin.RouterGroup) {
	taskConfigRouter := Router.Group("/task")
	taskConfigApi := v1.ApiGroupApp.ExampleApiGroup.TaskConfigApi
	subTaskConfigApi := v1.ApiGroupApp.ExampleApiGroup.SubTaskConfigApi
	{
		taskConfigRouter.GET("/list", taskConfigApi.LoadTaskConfigPage)
		taskConfigRouter.GET("/subtask/list", subTaskConfigApi.GetSubTaskConfigData)
		taskConfigRouter.POST("/mine/list", taskConfigApi.LoadOwnerTaskConfigPage)
		taskConfigRouter.POST("/approve/list", subTaskConfigApi.LoadSubTaskConfigPage)
		taskConfigRouter.POST("/log/list", taskConfigApi.LoadOwnerTaskConfigPage)
		taskConfigRouter.POST("/approve/uptStatus", subTaskConfigApi.UptSubTaskConfigData)
	}
}

File: web/src/api/task_config.js
import request from "@/request/index.js"

export const loadtaskconfig = ()=>{
    return request.get("/v1/task/list")
}

export const getSubTaskData = (id) => {
    return request.get(`/v1/task/subtask/list?id=${id}`);
};

export const loadOwnerTaskConfig = (owner)=>{
    return request.post("/v1/task/mine/list",{ owner: owner })
}

export const loadApproveTaskConfig = (owner)=>{
    return request.post("/v1/task/approve/list",{ owner: owner })
}

export const uptSubTaskConfigStatus = (id)=>{
    return request.post("/v1/task/approve/uptStatus",{ id: id })
}
